hw/ip/clkmgr/data/clkmgr.sv.tpl

```diff
@@ -285,6 +290,42 @@
 
   assign pwr_o.clk_status = clk_status;
 
+  ////////////////////////////////////////////////////
+  // Clock Measurement for the roots
+  ////////////////////////////////////////////////////
+
+<% aon_freq = clocks.all_srcs['aon'].freq %>\
+% for src in typed_clocks.rg_srcs:
+  logic ${src}_meas_valid;
+  logic ${src}_fast_err;
+  logic ${src}_slow_err;
+  <%
+   freq = clocks.all_srcs[src].freq
+   cnt = int(freq*2 / aon_freq)
+  %>\
+  prim_clock_meas #(
+    .Cnt(${cnt}),
+    .RefCnt(1)
```

Comment:
I think we may need to implement these counters with an expected precision. The slowest clock (io_div4) will have a precision of about 1/120. Is that enough? If not, RefCnt should be larger than 1.

Reply:
i personally think it's enough, the idea is really to capture gross deviations.  So using the 120 as an example, my thinking is that we would give it approximately a range of 118 ~ 122 or so (with the understanding that the delta may result from synchronization.  Of course, if you feel strongly it should be longer, we can either re-parameterize this, or add a programmable reference count (I prefer the former).

Reply:
My point is that we need to know the target precision, and that should determine the number of bits for the counters. The AST spec is the only place I recall that has any mention of the acceptable frequency deviation, but there are lots of TBDs.

Reply:
yes as mentioned, this is in my mind mainly to deal with gross deviations. We are not trying to deal with the deviations the oscillators themselves will natively have.